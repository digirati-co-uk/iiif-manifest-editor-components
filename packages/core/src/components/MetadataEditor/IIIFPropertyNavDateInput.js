import * as React from 'react';
import * as PropTypes from 'prop-types';
import IIIFInputField from '../fields/IIIFInputField/IIIFInputField';

const IIIFPropertyNavDateInput = ({
  labels,
  update,
  targetEntity,
  classes,
  type
}) => (
  <IIIFInputField
    label={labels[`${type}.NavDate`] || 'Nav Date'}
    type="datetime-local"
    className={classes.textField}
    value={targetEntity.navDate || ''}
    onChange={ev => update(
      targetEntity,
      'navDate',
      null,
      ev.target.value
    )}
    InputLabelProps={{
      shrink: true,
    }}
  />
);

IIIFPropertyNavDateInput.propTypes ={
  labels: PropTypes.object,
  update: PropTypes.func,
  targetEntity: PropTypes.object,
  classes: PropTypes.any,
  type: PropTypes.string,
};

export default IIIFPropertyNavDateInput;
