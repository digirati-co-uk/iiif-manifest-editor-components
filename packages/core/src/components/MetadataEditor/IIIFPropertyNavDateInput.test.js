import * as React from 'react';
import IIIFPropertyNavDateInput from './IIIFPropertyNavDateInput';

describe("IIIFPropertyNavDateInput", () => {
  it('Renders IIIFPropertyNavDateInput', () => {
    const handleUpdate = jest.fn();
    const target = {
      behavior: [],
    }
    const languageCode = 'en';
    const type = 'Manifest';
    const labels = {};
    const classes = {};
    const renderedInput = render(
      <IIIFPropertyNavDateInput
        labels={labels}
        update={handleUpdate}
        targetEntity={target}
        lang={languageCode}
        type={type}
        classes={classes}
      />
    );
    expect(renderedInput).toMatchSnapshot();
  });
  it('Responds on change', () => {
    const handleUpdate = jest.fn();
    const target = {
      behavior: [],
    }
    const languageCode = 'en';
    const type = 'Manifest';
    const labels = {};
    const classes = {};
    const wrappedInput = mount(
      <IIIFPropertyNavDateInput
        labels={labels}
        update={handleUpdate}
        targetEntity={target}
        lang={languageCode}
        type={type}
        classes={classes}
      />
    );
    wrappedInput.find('IIIFInputField').simulate('change', 'test');
    return (new Promise((resolve) => {
      setTimeout(() => {
        resolve()
      }, 1001);
    })).then(()=>{
      expect(handleUpdate.mock.calls.length).toBe(1);      
    });
  });
});