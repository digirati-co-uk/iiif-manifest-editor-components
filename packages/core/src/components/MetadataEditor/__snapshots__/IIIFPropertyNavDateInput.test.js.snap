// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IIIFPropertyNavDateInput Renders IIIFPropertyNavDateInput 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "MuiFormControl-root-1 MuiFormControl-marginDense-3",
    },
    "children": Array [
      Object {
        "attribs": Object {
          "class": "MuiFormLabel-root-16 MuiInputLabel-root-5 MuiInputLabel-formControl-10 MuiInputLabel-animated-13 MuiInputLabel-shrink-12 MuiInputLabel-marginDense-11 MuiInputLabel-outlined-15",
          "data-shrink": "true",
        },
        "children": Array [
          Object {
            "data": "Nav Date",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "text",
          },
        ],
        "name": "label",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Object {
          "attribs": Object {
            "class": "MuiInputBase-root-36 MuiOutlinedInput-root-23 MuiInputBase-formControl-37 MuiInputBase-marginDense-43",
          },
          "children": Array [
            Object {
              "attribs": Object {
                "aria-hidden": "true",
                "class": "MuiPrivateNotchedOutline-root-53 MuiOutlinedInput-notchedOutline-30",
                "style": "padding-left:8px",
              },
              "children": Array [
                Object {
                  "attribs": Object {
                    "class": "MuiPrivateNotchedOutline-legend-54",
                    "style": "width:0",
                  },
                  "children": Array [
                    Object {
                      "attribs": Object {},
                      "children": Array [
                        Object {
                          "data": "​",
                          "next": null,
                          "parent": [Circular],
                          "prev": null,
                          "type": "text",
                        },
                      ],
                      "name": "span",
                      "namespace": "http://www.w3.org/1999/xhtml",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "tag",
                      "x-attribsNamespace": Object {},
                      "x-attribsPrefix": Object {},
                    },
                  ],
                  "name": "legend",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                    "style": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                    "style": undefined,
                  },
                },
              ],
              "name": "fieldset",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": Object {
                "attribs": Object {
                  "aria-invalid": "false",
                  "class": "MuiInputBase-input-46 MuiOutlinedInput-input-31 MuiInputBase-inputType-49 MuiInputBase-inputMarginDense-47 MuiOutlinedInput-inputMarginDense-32",
                  "type": "datetime-local",
                  "value": "",
                },
                "children": Array [],
                "name": "input",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": [Circular],
                "type": "tag",
                "x-attribsNamespace": Object {
                  "aria-invalid": undefined,
                  "class": undefined,
                  "type": undefined,
                  "value": undefined,
                },
                "x-attribsPrefix": Object {
                  "aria-invalid": undefined,
                  "class": undefined,
                  "type": undefined,
                  "value": undefined,
                },
              },
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "aria-hidden": undefined,
                "class": undefined,
                "style": undefined,
              },
              "x-attribsPrefix": Object {
                "aria-hidden": undefined,
                "class": undefined,
                "style": undefined,
              },
            },
            Object {
              "attribs": Object {
                "aria-invalid": "false",
                "class": "MuiInputBase-input-46 MuiOutlinedInput-input-31 MuiInputBase-inputType-49 MuiInputBase-inputMarginDense-47 MuiOutlinedInput-inputMarginDense-32",
                "type": "datetime-local",
                "value": "",
              },
              "children": Array [],
              "name": "input",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": Object {
                "attribs": Object {
                  "aria-hidden": "true",
                  "class": "MuiPrivateNotchedOutline-root-53 MuiOutlinedInput-notchedOutline-30",
                  "style": "padding-left:8px",
                },
                "children": Array [
                  Object {
                    "attribs": Object {
                      "class": "MuiPrivateNotchedOutline-legend-54",
                      "style": "width:0",
                    },
                    "children": Array [
                      Object {
                        "attribs": Object {},
                        "children": Array [
                          Object {
                            "data": "​",
                            "next": null,
                            "parent": [Circular],
                            "prev": null,
                            "type": "text",
                          },
                        ],
                        "name": "span",
                        "namespace": "http://www.w3.org/1999/xhtml",
                        "next": null,
                        "parent": [Circular],
                        "prev": null,
                        "type": "tag",
                        "x-attribsNamespace": Object {},
                        "x-attribsPrefix": Object {},
                      },
                    ],
                    "name": "legend",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "class": undefined,
                      "style": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "class": undefined,
                      "style": undefined,
                    },
                  },
                ],
                "name": "fieldset",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": [Circular],
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "aria-hidden": undefined,
                  "class": undefined,
                  "style": undefined,
                },
                "x-attribsPrefix": Object {
                  "aria-hidden": undefined,
                  "class": undefined,
                  "style": undefined,
                },
              },
              "type": "tag",
              "x-attribsNamespace": Object {
                "aria-invalid": undefined,
                "class": undefined,
                "type": undefined,
                "value": undefined,
              },
              "x-attribsPrefix": Object {
                "aria-invalid": undefined,
                "class": undefined,
                "type": undefined,
                "value": undefined,
              },
            },
          ],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
          },
        },
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
          "data-shrink": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
          "data-shrink": undefined,
        },
      },
      Object {
        "attribs": Object {
          "class": "MuiInputBase-root-36 MuiOutlinedInput-root-23 MuiInputBase-formControl-37 MuiInputBase-marginDense-43",
        },
        "children": Array [
          Object {
            "attribs": Object {
              "aria-hidden": "true",
              "class": "MuiPrivateNotchedOutline-root-53 MuiOutlinedInput-notchedOutline-30",
              "style": "padding-left:8px",
            },
            "children": Array [
              Object {
                "attribs": Object {
                  "class": "MuiPrivateNotchedOutline-legend-54",
                  "style": "width:0",
                },
                "children": Array [
                  Object {
                    "attribs": Object {},
                    "children": Array [
                      Object {
                        "data": "​",
                        "next": null,
                        "parent": [Circular],
                        "prev": null,
                        "type": "text",
                      },
                    ],
                    "name": "span",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "tag",
                    "x-attribsNamespace": Object {},
                    "x-attribsPrefix": Object {},
                  },
                ],
                "name": "legend",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                  "style": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                  "style": undefined,
                },
              },
            ],
            "name": "fieldset",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": Object {
              "attribs": Object {
                "aria-invalid": "false",
                "class": "MuiInputBase-input-46 MuiOutlinedInput-input-31 MuiInputBase-inputType-49 MuiInputBase-inputMarginDense-47 MuiOutlinedInput-inputMarginDense-32",
                "type": "datetime-local",
                "value": "",
              },
              "children": Array [],
              "name": "input",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": [Circular],
              "type": "tag",
              "x-attribsNamespace": Object {
                "aria-invalid": undefined,
                "class": undefined,
                "type": undefined,
                "value": undefined,
              },
              "x-attribsPrefix": Object {
                "aria-invalid": undefined,
                "class": undefined,
                "type": undefined,
                "value": undefined,
              },
            },
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "aria-hidden": undefined,
              "class": undefined,
              "style": undefined,
            },
            "x-attribsPrefix": Object {
              "aria-hidden": undefined,
              "class": undefined,
              "style": undefined,
            },
          },
          Object {
            "attribs": Object {
              "aria-invalid": "false",
              "class": "MuiInputBase-input-46 MuiOutlinedInput-input-31 MuiInputBase-inputType-49 MuiInputBase-inputMarginDense-47 MuiOutlinedInput-inputMarginDense-32",
              "type": "datetime-local",
              "value": "",
            },
            "children": Array [],
            "name": "input",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": Object {
              "attribs": Object {
                "aria-hidden": "true",
                "class": "MuiPrivateNotchedOutline-root-53 MuiOutlinedInput-notchedOutline-30",
                "style": "padding-left:8px",
              },
              "children": Array [
                Object {
                  "attribs": Object {
                    "class": "MuiPrivateNotchedOutline-legend-54",
                    "style": "width:0",
                  },
                  "children": Array [
                    Object {
                      "attribs": Object {},
                      "children": Array [
                        Object {
                          "data": "​",
                          "next": null,
                          "parent": [Circular],
                          "prev": null,
                          "type": "text",
                        },
                      ],
                      "name": "span",
                      "namespace": "http://www.w3.org/1999/xhtml",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "tag",
                      "x-attribsNamespace": Object {},
                      "x-attribsPrefix": Object {},
                    },
                  ],
                  "name": "legend",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                    "style": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                    "style": undefined,
                  },
                },
              ],
              "name": "fieldset",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": [Circular],
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "aria-hidden": undefined,
                "class": undefined,
                "style": undefined,
              },
              "x-attribsPrefix": Object {
                "aria-hidden": undefined,
                "class": undefined,
                "style": undefined,
              },
            },
            "type": "tag",
            "x-attribsNamespace": Object {
              "aria-invalid": undefined,
              "class": undefined,
              "type": undefined,
              "value": undefined,
            },
            "x-attribsPrefix": Object {
              "aria-invalid": undefined,
              "class": undefined,
              "type": undefined,
              "value": undefined,
            },
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": Object {
          "attribs": Object {
            "class": "MuiFormLabel-root-16 MuiInputLabel-root-5 MuiInputLabel-formControl-10 MuiInputLabel-animated-13 MuiInputLabel-shrink-12 MuiInputLabel-marginDense-11 MuiInputLabel-outlined-15",
            "data-shrink": "true",
          },
          "children": Array [
            Object {
              "data": "Nav Date",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
          ],
          "name": "label",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
            "data-shrink": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
            "data-shrink": undefined,
          },
        },
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
